RecoEgamma/PhotonIdentification/plugins/PhotonIDValueMapProducer.cc

```diff
@@ -546,17 +605,35 @@ PhotonIDValueMapProducer::candidatePdgId(const edm::Ptr<reco::Candidate> candida
   return thisCandidateType;
 }
 
-std::pair<float,float>
-PhotonIDValueMapProducer::getTrackDxyDz(const edm::Ptr<reco::Candidate> & candidate, const reco::Particle::Point & vtxpos, bool isAOD) {
+const reco::Track* 
+PhotonIDValueMapProducer::getTrackPointer(const edm::Ptr<reco::Candidate> candidate, bool isAOD){
 
+  const reco::Track* theTrack = nullptr;
+  if( isAOD )
+    theTrack = &*( ((const recoCandPtr) candidate)->trackRef());
+  else
+    theTrack = &( ((const patCandPtr) candidate)->pseudoTrack());
+
+  return theTrack;
+}
+
+void PhotonIDValueMapProducer::getImpactParameters(const edm::Ptr<reco::Candidate> candidate,
+                                                   bool isAOD, const reco::Vertex pv,
+                                                   float &dxy, float &dz){
+
+  dxy=-999;
+  dz=-999;
   if( isAOD ) {
-    const reco::Track & theTrack = *recoCandPtr(candidate)->trackRef();
-    return std::make_pair(theTrack.dxy(vtxpos),theTrack.dz(vtxpos));
+    const reco::Track *theTrack = &*( ((const recoCandPtr) candidate)->trackRef());
+    dxy = theTrack->dxy(pv.position());
+    dz  = theTrack->dz(pv.position());
   } else {
-    const pat::PackedCandidate & theCand = *(patCandPtr(candidate));
-    return std::make_pair(theCand.dxy(vtxpos),theCand.dz(vtxpos));
+    dxy = ((const patCandPtr) candidate)->dxy(pv.position());
+    dz = ((const patCandPtr) candidate)->dz(pv.position());
+
   }
 
+  return;
```

Comment:
is this needed?

Reply:
Changes implemented here:
https://github.com/lsoffi/cmssw/blob/CMSSW_9_4_0_pre3_TnP/RecoEgamma/PhotonIdentification/plugins/PhotonIDValueMapProducer.cc